src/components/Table/BaseTable.tsx

```diff
@@ -51,127 +55,171 @@ type SpruceTableProps = {
-              return (
-                <HeaderCell
-                  key={header.id}
-                  header={header}
-                  style={meta?.width && { width: columnDef?.meta?.width }}
-                >
-                  {flexRender(columnDef.header, header.getContext())}
-                  {header.column.getCanFilter() &&
-                    (meta?.treeSelect ? (
-                      <TableFilterPopover
-                        data-cy={meta.treeSelect?.["data-cy"]}
-                        onConfirm={(value) =>
-                          header.column.setFilterValue(value)
-                        }
-                        options={
-                          meta.treeSelect?.filterOptions
-                            ? meta.treeSelect.options.filter(
-                                ({ value }) =>
-                                  !!header.column
-                                    .getFacetedUniqueValues()
-                                    .get(value)
-                              )
-                            : meta.treeSelect.options
-                        }
-                        value={
-                          (header?.column?.getFilterValue() as string[]) ?? []
-                        }
-                      />
-                    ) : (
-                      <TableSearchPopover
-                        data-cy={meta?.search?.["data-cy"]}
-                        onConfirm={(value) =>
-                          header.column.setFilterValue(value)
-                        }
-                        placeholder={meta?.search?.placeholder}
-                        value={
-                          (header?.column?.getFilterValue() as string) ?? ""
-                        }
-                      />
-                    ))}
-                </HeaderCell>
-              );
-            })}
-          </HeaderRow>
-        ))}
-      </TableHead>
-      <TableBody>
-        {loading && (
-          <TableLoader
-            numColumns={table.getAllColumns().length}
-            numRows={loadingRows}
-          />
-        )}
-        {table.getRowModel().rows.map((row) => (
-          <Row
-            key={row.id}
-            row={row}
-            data-cy="leafygreen-table-row"
-            className={css`
-              &[aria-hidden="false"] td > div {
-                max-height: unset;
-              }
-            `}
-          >
-            {row.getVisibleCells().map((cell) => (
-              <Cell key={cell.id} style={{ padding: `${size.xxs} 2px` }}>
-                {flexRender(cell.column.columnDef.cell, cell.getContext())}
-              </Cell>
+export const BaseTable = forwardRef(
+  (
+    {
+      className,
+      "data-cy-row": dataCyRow,
+      "data-cy-table": dataCyTable,
+      emptyComponent,
+      loading,
+      loadingRows = 5,
+      table,
+      ...args
+    }: SpruceTableProps & TableProps<any>,
+    ref: ForwardedRef<HTMLDivElement>
```

Comment:
What exactly is the need for passing the ref as a component on the prop? As best I can tell, LG [supports](https://github.com/mongodb/leafygreen-ui/tree/main/packages/table#usevirtualscrolling) virtualized scrolling, and already requires a ref via [the `containerRef` prop](https://github.com/mongodb/leafygreen-ui/tree/main/packages/table#useleafygreentable), and so we may not need to deal with the ref ourselves.

Reply:
The ref wasn't actually being passed to the underlying table component. [forwardRef](https://react.dev/reference/react/forwardRef) is required in order for the ref to be passed to the component. Without it the `containerRef` prop wouldn't have any effect and it is required for the leafygreen tables virtualization functionality.